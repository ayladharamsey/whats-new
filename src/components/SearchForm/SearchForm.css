form {
    width: 80vw;
    font-family: Roboto;
    margin-top: 20px;
}

input {
    width: 350px;
    height: 30px;
    border-style: solid;
    border-color: #C2D68B;
    border-width: 1px;
    border-radius: 5px;
    background-color: #F0F0F0;
}

button {
    width: 70px;
    height: 30px;
    border-color: #B7E6D6;
    background-color:#F0F0F0;
    font-size: .7em;
    border-radius: 5px;
}