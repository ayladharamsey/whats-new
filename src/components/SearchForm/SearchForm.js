import React, { Component } from 'react';
import './SearchForm.css';

class SearchForm extends Component {
    constructor() {
        super();
        this.state = {
            query: ""
        }
    };

    submitQuery = event => {
        this.setState({[event.target.name]: [event.target.value]})
    };

    handleSumbit = event => {
        event.preventDefault();
        const search = this.state.query;
        this.setState({query:""});
        this.props.searchArticles(search);
    };

    reset = () => {
        this.setState({
          query: ""
        })
      };

    render() {
        return (
            <form>
                <input type="text" name="query" value={this.state.query} onChange={this.submitQuery} />
                <button onClick={this.handleSumbit}>Search</button>
                <button onClick={this.reset}> Reset </button>
            </form>
        )

    };
}

export default SearchForm;