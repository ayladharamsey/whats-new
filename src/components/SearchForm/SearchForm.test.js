import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm'

describe('SearchForm', () => {
    let wrapper, mockEvent, searchArticlesMock; 

    beforeEach(() => {        
        mockEvent = { target: { name: 'test', value: 'testing'} }
        searchArticlesMock=jest.fn();
        wrapper = shallow(<SearchForm searchArticles = {searchArticlesMock}/>)
    });

    it('should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should update the state\'s query', () => {
        expect(wrapper.state('query')).toEqual('');
        wrapper.instance().submitQuery(mockEvent);
        expect(wrapper.state('selectedArticle')).toEqual(mockEvent.value)
    });

    it('should reset state back to undefined after handling submit', () => {
        let mockPreventDefault = jest.fn()
        let mockEvent = {
            preventDefault : mockPreventDefault
        }
        wrapper.instance().handleSumbit(mockEvent)

        expect(mockPreventDefault).toHaveBeenCalled();
        wrapper.instance().searchArticles = jest.fn();
        expect(wrapper.state()).toEqual({query: ''});
    });

    it.skip('should query state upon typing', () => {  
        wrapper.instance().submitQuery(mockEvent)      
        wrapper.find('input').simulate('change')
        expect(wrapper.state('query')).toEqual({query: 'testing'})
    });


});