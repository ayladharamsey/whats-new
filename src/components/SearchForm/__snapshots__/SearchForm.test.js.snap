// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot 1`] = `
<form>
  <input
    name="query"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Search
  </button>
  <button
    onClick={[Function]}
  >
     Reset 
  </button>
</form>
`;
